### PR TITLE
Support tailing the cloudformation event stream when building and destroying stacks

### PR DESCRIPTION
**Pull request by mhahn**
output for building looks like:

```
$ stacker build -p BaseDomain=mike.com -p SshKeyName=circle conf/stage.env conf/example.yaml --tail
[2015-09-08T19:36:33] Executing pre_build hooks: stacker.hooks.route53.create_domain
[2015-09-08T19:36:33] Launching stacks: stacker-dev-vpc, stacker-dev-bastion
[2015-09-08T19:36:33] Tailing stack: stacker-dev-vpc
[2015-09-08T19:36:34] Launching stack stacker-dev-vpc now.
[2015-09-08T19:36:35] Stack stacker-dev-vpc not found, creating.
[2015-09-08T19:36:35] Tailing stack: stacker-dev-vpc
[2015-09-08T19:36:36] Tailing stack: stacker-dev-vpc
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPC
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::InternetGateway
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::DHCPOptions
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::InternetGateway Resource creation Initiated
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::DHCPOptions Resource creation Initiated
[2015-09-08T19:36:52] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPC Resource creation Initiated
[2015-09-08T19:37:07] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::InternetGateway
[2015-09-08T19:37:07] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::DHCPOptions
[2015-09-08T19:37:07] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::VPC
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPCGatewayAttachment
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPCDHCPOptionsAssociation
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPCGatewayAttachment Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::RouteTable Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::VPCDHCPOptionsAssociation Resource creation Initiated
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::VPCDHCPOptionsAssociation
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::RouteTable
[2015-09-08T19:37:12] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::NetworkAcl
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::NetworkAcl Resource creation Initiated
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route Resource creation Initiated
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route Resource creation Initiated
[2015-09-08T19:37:18] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::NetworkAcl
[2015-09-08T19:37:22] Plan Status:
[2015-09-08T19:37:22]   - step "stacker-dev-vpc": submitted
[2015-09-08T19:37:22]   - step "stacker-dev-bastion": pending
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup Resource creation Initiated
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::VPCGatewayAttachment
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SecurityGroup
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet
[2015-09-08T19:37:28] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet Resource creation Initiated
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet Resource creation Initiated
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet Resource creation Initiated
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Subnet Resource creation Initiated
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Route
[2015-09-08T19:37:34] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Route
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup Resource creation Initiated
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Subnet
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Subnet
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Subnet
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SecurityGroup
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Subnet
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Instance
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Instance
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation Resource creation Initiated
[2015-09-08T19:37:49] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation Resource creation Initiated
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Instance Resource creation Initiated
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Instance Resource creation Initiated
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation Resource creation Initiated
[2015-09-08T19:37:54] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::SubnetRouteTableAssociation Resource creation Initiated
[2015-09-08T19:38:10] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:38:10] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:38:10] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:38:10] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::SubnetRouteTableAssociation
[2015-09-08T19:38:13] Plan Status:
[2015-09-08T19:38:13]   - step "stacker-dev-vpc": submitted
[2015-09-08T19:38:13]   - step "stacker-dev-bastion": pending
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Instance
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Instance
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::EIP
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::EIP
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::EIP Resource creation Initiated
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::EIP Resource creation Initiated
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route Resource creation Initiated
[2015-09-08T19:38:41] [stacker-dev-vpc] CREATE_IN_PROGRESS AWS::EC2::Route Resource creation Initiated
[2015-09-08T19:38:57] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Route
[2015-09-08T19:38:57] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::Route
[2015-09-08T19:39:05] Plan Status:
[2015-09-08T19:39:05]   - step "stacker-dev-vpc": submitted
[2015-09-08T19:39:05]   - step "stacker-dev-bastion": pending
[2015-09-08T19:39:13] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::EIP
[2015-09-08T19:39:13] [stacker-dev-vpc] CREATE_COMPLETE AWS::EC2::EIP
[2015-09-08T19:39:15] Tailing stack: stacker-dev-bastion
[2015-09-08T19:39:15] Launching stack stacker-dev-bastion now.
[2015-09-08T19:39:17] Tailing stack: stacker-dev-bastion
[2015-09-08T19:39:17] Stack stacker-dev-bastion not found, creating.
[2015-09-08T19:39:18] Tailing stack: stacker-dev-bastion
[2015-09-08T19:39:19] Tailing stack: stacker-dev-bastion
[2015-09-08T19:39:51] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup
[2015-09-08T19:40:00] Plan Status:
[2015-09-08T19:40:00]   - step "stacker-dev-vpc": complete
[2015-09-08T19:40:00]   - step "stacker-dev-bastion": submitted
[2015-09-08T19:40:06] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::EC2::SecurityGroup Resource creation Initiated
[2015-09-08T19:40:06] [stacker-dev-bastion] CREATE_COMPLETE AWS::EC2::SecurityGroup
[2015-09-08T19:40:06] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::AutoScaling::LaunchConfiguration
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::EC2::SecurityGroupIngress
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::EC2::SecurityGroupIngress Resource creation Initiated
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_COMPLETE AWS::EC2::SecurityGroupIngress
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::AutoScaling::LaunchConfiguration Resource creation Initiated
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_COMPLETE AWS::AutoScaling::LaunchConfiguration
[2015-09-08T19:40:11] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::AutoScaling::AutoScalingGroup
[2015-09-08T19:40:16] [stacker-dev-bastion] CREATE_IN_PROGRESS AWS::AutoScaling::AutoScalingGroup Resource creation Initiated
[2015-09-08T19:40:53] Plan Status:
[2015-09-08T19:40:53]   - step "stacker-dev-vpc": complete
[2015-09-08T19:40:53]   - step "stacker-dev-bastion": submitted
[2015-09-08T19:41:19] [stacker-dev-bastion] CREATE_COMPLETE AWS::AutoScaling::AutoScalingGroup
[2015-09-08T19:41:19] [stacker-dev-bastion] CREATE_COMPLETE AWS::CloudFormation::Stack
[2015-09-08T19:41:19] Plan Status:
[2015-09-08T19:41:19]   - step "stacker-dev-vpc": complete
[2015-09-08T19:41:19]   - step "stacker-dev-bastion": complete
```

### CODE REVIEW

**Comment by mhahn** [comment]
this depends on this PR for troposphere: https://github.com/cloudtools/troposphere/pull/315

**Comment by mhahn** [comment]
@phobologic referenced latest troposphere release, should be good to go

File: stacker/plan.py [hunk 1]
```diff
@@ -182,14 +191,30 @@ def _single_run(self):
                 )
                 continue
 
+            if not step.done and self._watch_func and step_name not in self._watchers:
```
Comment (phobologic) [inline stacker/plan.py]: This seems like it could use some tests.  A little nervous about introducing multiprocessing without any.

Comment (mhahn) [inline stacker/plan.py]: added a test in:  a089dc3

i've also been using this for a couple of months now

**Pull request merged**